src/types.ts

```diff
@@ -48,9 +48,28 @@ export interface Component {
   type: string;
 }
 
-export interface PrefabComponent {
-  actions?: PrefabAction[];
+export const isComponentTypePrefab = (
+  component: PrefabComponent,
+): component is ComponentTypePrefab =>
+  typeof component.type === 'undefined' || component.type === 'COMPONENT';
+export const isPartialTypePrefab = (
+  component: PrefabComponent,
+): component is PartialTypePrefab => component.type === 'PARTIAL';
+
+export type PrefabComponent = ComponentTypePrefab | PartialTypePrefab;
+
+export type ComponentTypePrefab = BasePrefab & {
```

Comment:
Why extend `BasePrefab`? That doesn't make any sense. I don't think you want to be able to add actions, interactions, styles and options to a `PrefabPartial`? 😛

You probably want to rename `BasePrefab` to `Prefab` and make sure that the `structure` and `descendants` accept a list of objects, e.g. `PrefabReference[]` (`type PrefabReference = PrefabComponent | PrefabPartial`)

Reply:
We have removed the BasePrefab and made the PrefabComponent to default. The PrefabComponent ha an array of either PrefabComponents, or a PrefabPartial.